website/pages/docs/k8s/connect/ingress-gateways.mdx

```diff
@@ -5,4 +5,174 @@ sidebar_title: Ingress Gateways
+
+-> If TLS is enabled set:
+```shell-session
+$ export CONSUL_HTTP_ADDR=https://localhost:8501
+```
+
+-> If ACLs are enabled set :
+```shell-session
+$ export CONSUL_HTTP_TOKEN=$(kubectl get secret consul-bootstrap-acl-token -o jsonpath={.data.token} | base64 -D)
+$ export CONSUL_HTTP_SSL_VERIFY=false
+```
+
+### Submitting an Ingress Gateway configuration entry
+
+Now that you have access to the Consul CLI you are able to submit an ingress gateway configuration entry.
+Here is an ingress gateway configuration.
+
+```shell
+Kind = "ingress-gateway"
+Name = "ingress-gateway"
+
+Listeners = [
+ {
+   Port = 8080
+   Protocol = "http"
+   Services = [
+     {
+       Name = "static-server"
+     }
+   ]
+ }
+]
+```
+
+Submit the ingress gateway configuration with the Consul CLI using this command.
+
+```shell-session
+$ consul config write ingress-gateway.hcl
+```
+
+-> You can confirm the ingress gateways have been configured as expected by looking at the Consul UI: http://localhost:8500/ui/dc1/services/ingress-gateway/
+
+-> If TLS is enabled use port 8501
+
+## Defining an Intention
+
+If ACLs are enabled, you must define an [intention](/docs/commands/intention) to allow the ingress gateway to access the upstream services defined in the config entries.
+For detailed instructions on how to configure zero-trust networking with intentions please refer to this [guide](https://learn.hashicorp.com/consul/gs-consul-service-mesh/network-security-with-consul-service-mesh).
+
+## Deploying your application to Kubernetes
+Now you will deploy a sample application which echos “hello world”
+
+```shell
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: static-server
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-server
+  annotations:
+    'consul.hashicorp.com/connect-inject': 'true'
+spec:
+  containers:
+    # This name will be the service name in Consul.
+    - name: static-server
+      image: hashicorp/http-echo:latest
+      args:
+        - -text="hello world"
+        - -listen=:8080
+      ports:
+        - containerPort: 8080
+          name: http
+    # If ACLs are enabled, the serviceAccountName must match the Consul service name.
+  serviceAccountName: static-server
+```
+
+```shell-session
+kubectl apply -f static-server.yaml
+```
+
+## Connecting to your application
+
+You can validate the service is running and registered in the Consul UI by navigating to
+`http://localhost:8500/ui/dc1/services/static-server/instances`
+
+-> If TLS is enabled use port 8501
+
+You can also validate the connectivity of the application using a curl command
+```shell-session
+EXTERNAL_IP=$(kubectl get services | grep ingress-gateway | awk ‘{print $4}’)
+curl -H "Host: static-server.ingress.consul" $EXTERNAL_IP:8080
+```
+
+-> *Note:* Please be sure to delete the application and services created here as they represent a security risk through
+leaving an open and unathenticated load balancer alive in your cluster.
```

Comment:
```suggestion
leaving an open and unauthenticated load balancer alive in your cluster.
```
Can you add

To delete the ingress gateway, set `enabled` to `false` in your Helm configuration:
```
global:
  name: consul
connectInject:
  enabled: true
ingressGateways:
  enabled: false # Set to false
  gateways:
    - name: ingress-gateway
      replicas: 2
      service:
       type: LoadBalancer
```

And run Helm upgrade:
```
$ helm upgrade consul hashicorp/consul -f config.yaml
```